### PR TITLE
Add advisers to the dashboard

### PR DESCRIPTION
**Pull request by neoeno**
image

The right hand column is the addition here. The left hand column needs addressing as it currently goes on forever.

### CODE REVIEW

File: app/helpers/dashboard/dashboard_helper.rb [hunk 1]
```diff
@@ -22,5 +22,18 @@ def add_adviser_button(firm:)
         concat content_tag(:span, sr_label, class: 'visually-hidden')
       end
     end
+
+    def number_of_advisers(firm)
+      all_advisers_for_firm_and_trading_names(firm).size
+    end
+
+    def most_recently_edited_advisers(firm, limit = 3)
```
Comment (asmega) [inline app/helpers/dashboard/dashboard_helper.rb]: personally i prefer an options hash as I think it's more flexible
```
def most_recently_edited_advisers(firm, options = {})
  limit = options[:limit] || 3
end
```

**Comment by benbarnett** [comment]
:+1: for frontend

File: app/helpers/dashboard/dashboard_helper.rb (hunk 1 above)
Comment (neoeno) [inline app/helpers/dashboard/dashboard_helper.rb]: How do you feel about keyword args?

Comment (paulanthonywilson) [inline app/helpers/dashboard/dashboard_helper.rb]: I think keyword args are the way forward in this case.

```
def most_recently_edited_advisers(firm, limit: 3)
```


Much cleaner. (Aside, I've been using them a lot on the a different project and I was thinking of proposing "Keyword Arg Better Practice" as a Fireside discussion.)

Comment (asmega) [inline app/helpers/dashboard/dashboard_helper.rb]: sounds good

**Comment by rnalexander** [comment]
:+1: for code

**Comment by darokel** [comment]
It feels a little funny that view helpers are calling queries https://github.com/moneyadviceservice/rad/commit/b374664312cdf8c74dc441f3c60098b743227a2c, that's not something they should be concerned with. Ideally, they should just receive variables to use.

**Comment by neoeno** [comment]
Appreciated. It should live on `Firm`

**Comment by neoeno** [comment]
Fair few changes here. Moved the helper methods that were getting advisers onto `Adviser` in `mas-rad_core` and moved the tests etc. Called it through from the controllers.

This uncovered a lack of test coverage — our feature tests for the dashboard didn't include the advisers list. So I added that, then DRY'd it up a little.

So this'll need rethumbing I guess!

**Comment by darokel** [comment]
:+1: :ship:

**Comment by munckymagik** [comment]
Have reviewed the changes since Ryan's thumb, and I think :+1:

**Pull request merged**